### ISSUE TITLE
A false positive about EI_EXPOSE_REP2

### ISSUE THREAD

**Issue by Belle-PL** [spotbugs/spotbugs#2335]
In the following code example, SpotBugs reports a EI_EXPOSE_REP2 warning at line 2, but this is a false positive. No internal representation is exposed. 
```java
public class C {
  public class C1 {  // report a warning, a false positive
    public Object[] getObjects() {
      return new Object[5];
    }
  }
  public Object[] addCollection() {
    return new Object[10];
  }
}
```
However, the following code example does not have this warning and they are equivalent.
```java
public class C {
  public Object[] getObjects() {
    return new Object[5];
  }
  public Object[] addCollection() {
    return new Object[10];
  }
}
```

SpotBugs version: 4.7.3

**Comment by gtoison** [comment]
This is the same issue as #1797 the `addCollection()` method is assumed to be a "setter" because its name starts with "add"

**Comment by gtoison** [comment]
Closing this as a duplicate of #1797
